src/storage/mvcc/reader/forward_scanner.rs

```diff
@@ -167,8 +167,28 @@ impl<S: Snapshot> ForwardScanner<S> {
             self.is_started = true;
         }
 
+        // The general idea is to simultaneously step write cursor and lock cursor.
```

Comment:
I think you can make `read_next` much more simple when you check all locks in first call of `read_next`.

Reply:
how can we gracefully handle cop limit?